### PR TITLE
Fix/dropdown UI

### PR DESCRIPTION
**Pull request by musharaf-nocodb**
## Change Summary

Provide summary of changes with issue number if any.

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

hover effect in dropdown's is not consistent. some of the places that need correction are

 

- [x] filter
         
         
Screenshot 2023-12-07 at 2 21 15 PM

- [x]  sort
- [x]  group-by
- [x]  row height
- [x]  field context menu
- [x]  add/edit modal, fields drop down

Screenshot 2023-12-07 at 5 52 41 PM

---
- [x]  i18n language menu (OSS)
         
Screenshot 2023-12-11 at 8 23 49 AM

  
Screenshot 2023-12-11 at 8 25 37 AM

---
- [x]  global search modal

        
Screenshot 2023-12-08 at 10 19 54 AM

---

- [x]  webhook dropdown (event, url, action, headers, etc)
- [x]  API sub-type modal
         
         
Screenshot 2023-12-07 at 4 42 32 PM

---

- [x]  Team & Settings, User management (invite user, role drop down) (OSS)
          
Screenshot 2023-12-07 at 2 15 53 PM
Screenshot 2023-12-07 at 2 18 05 PM

---
- [x]  Data source (database type, ssl mode, inflection menu)
- [x]  new record

 
## Additional information / screenshots (optional)

Desired Behavior

grey background with rounded rectangle + edge padding

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by dstala** [comment]
Screenshot 2023-12-08 at 2 09 18 PM


- [ ] Current selected option when a drop down opens : highlight in blue as in case of language.
  - [ ] Collaborative mode / Locked mode
  - [ ] Fields add / edit modal
  - [ ] Global search
  - [ ] Record height
  - [ ] Details tab, APIs, sub option
  - [ ] New data source modal - multiple fields
- [ ]

**Comment by mustafapc19** [comment]
Search text seems to bigger than other texts
image
image

**Comment by musharaf-nocodb** [comment]
> * Search text seems to bigger than other texts
> 
> image image

**My CHANGE**

Screenshot 2023-12-11 at 9 55 18 AM

**DEVELOP BRANCH**

Screenshot 2023-12-11 at 9 55 10 AM

the text in my pr and in develop branch seems to be same.


> * Doesn't seem to have shadow
> 
> image


**FIXED IN BELOW COMMIT**

[fix : fixed shadow for dropdown](https://github.com/nocodb/nocodb/pull/7181/commits/78fc77769e0792146b4ea7181ac246337fdbf8ff) 
[78fc777](https://github.com/nocodb/nocodb/pull/7181/commits/78fc77769e0792146b4ea7181ac246337fdbf8ff)

> * This dropdown doesn't seem to support 'up' and 'down' btn to navigate the dropdown
> 
> image

**Comment by dstala** [comment]
- [ ] field add modal, option to add `JSON` not accessible
Screenshot 2023-12-11 at 1 30 43 PM

**Review by dstala — APPROVED** [review]

**Review by wingkwong — CHANGES_REQUESTED** [review]
run `pnpm run lint` on both `nc-gui` and `nocodb`

File: packages/nc-gui/components/nc/SelectOption.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+<script lang="ts" setup>
+interface Props {
+  isSelected?: boolean
+  value?: string
+}
+
+const props = defineProps<Props>()
+const isSelected = computed(() => props.isSelected)
+const value = computed(() => props.value)
+console.log(props.isSelected, isSelected)
```
Comment (wingkwong) [inline packages/nc-gui/components/nc/SelectOption.vue:20]: remove that

File: packages/nc-gui/components/nc/SelectOption.vue [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<template>
```
Comment (wingkwong) [inline packages/nc-gui/components/nc/SelectOption.vue:1]: where do we use this component?

File: packages/nc-gui/components/nc/SelectOption.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        <slot />
+      </div>
+      <component v-if="isSelected" id="nc-selected-item-icon" class="text-primary w-6 h-6" :is="iconMap.check" />
+    </div>
+  </a-select-option>
+</template>
+<script lang="ts" setup>
+interface Props {
+  isSelected?: boolean
+  value?: string
```
Comment (wingkwong) [inline packages/nc-gui/components/nc/SelectOption.vue:14]: value shouldn't be optional imo

File: packages/nc-gui/assets/style.scss [hunk 4]
```diff
@@ -36,6 +36,9 @@ body {
   overflow-y: visible !important;
 }
 
+.rc-virtual-list-holder-inner{
```
Comment (wingkwong) [inline packages/nc-gui/assets/style.scss:39]: add a space before `{`

File: packages/nc-gui/assets/style.scss [hunk 5]
```diff
@@ -505,8 +508,18 @@ a {
...
 
+.ant-select-item-option-selected:not(.ant-select-item-option-disabled):hover,
 .ant-select-item-option-active:not(.ant-select-item-option-disabled) {
-  @apply bg-primary bg-opacity-20;
+  @apply bg-gray-300 bg-opacity-20;
+}
+
+/* Hide the element with id nc-selected-item-icon */
+.ant-select-selection-item #nc-selected-item-icon {
+  display: none;
```
Comment (wingkwong) [inline packages/nc-gui/assets/style.scss:518]:
```
@apply hidden;
```

File: packages/nc-gui/components/nc/SelectOption.vue (hunk 3 above)
Comment (musharaf-nocodb) [reply packages/nc-gui/components/nc/SelectOption.vue:4]: ignore this component

Comment (musharaf-nocodb) [reply packages/nc-gui/components/nc/SelectOption.vue:4]: not sure how it got pushed

File: packages/nc-gui/assets/style.scss [hunk 6]
```diff
@@ -592,6 +605,7 @@ a {
 
 .ant-popover-inner {
   padding: 0 !important;
+  @apply rounded-lg
```
Comment (wingkwong) [inline packages/nc-gui/assets/style.scss:608]: merge with L607 also, i.e.

```
@apply rouded-lg !p-0;
```

File: packages/nc-gui/components/nc/SelectOption.vue (hunk 3 above)
Comment (musharaf-nocodb) [reply packages/nc-gui/components/nc/SelectOption.vue:4]: removing it

File: packages/nc-gui/components/smartsheet/column/LookupOptions.vue [hunk 7]
```diff
@@ -83,15 +83,27 @@ const cellIcon = (column: ColumnType) =>
       <a-form-item class="flex w-1/2 pb-2" :label="$t('labels.links')" v-bind="validateInfos.fk_relation_column_id">
         <a-select
           v-model:value="vModel.fk_relation_column_id"
-          dropdown-class-name="!w-64 nc-dropdown-relation-table"
+          dropdown-class-name="!w-64 !rounded-md nc-dropdown-relation-table "
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/column/LookupOptions.vue:86]: unexpected space at the end

File: packages/nc-gui/components/nc/SelectOption.vue (hunk 2 above)
Comment (musharaf-nocodb) [reply packages/nc-gui/components/nc/SelectOption.vue]: **Removed in below commit**

[fix : removed SelectOption component](https://github.com/nocodb/nocodb/pull/7181/commits/0ef580d48e9c98ce3c4cc2e3b4cdfd37f9637860)
[0ef580d](https://github.com/nocodb/nocodb/pull/7181/commits/0ef580d48e9c98ce3c4cc2e3b4cdfd37f9637860)

File: packages/nc-gui/components/nc/SelectOption.vue (hunk 1 above)
Comment (musharaf-nocodb) [reply packages/nc-gui/components/nc/SelectOption.vue]: **REMOVED in BELOW COMMIT**

[fix : removed SelectOption component](https://github.com/nocodb/nocodb/pull/7181/commits/0ef580d48e9c98ce3c4cc2e3b4cdfd37f9637860)
[0ef580d](https://github.com/nocodb/nocodb/pull/7181/commits/0ef580d48e9c98ce3c4cc2e3b4cdfd37f9637860)

File: packages/nc-gui/components/smartsheet/column/LookupOptions.vue (hunk 7 above)
Comment (musharaf-nocodb) [reply packages/nc-gui/components/smartsheet/column/LookupOptions.vue]: **Fixed in below commit**

[fix : removed unwanted spaces](https://github.com/nocodb/nocodb/pull/7181/commits/4b8ee083f5689fd543f5a727c98a384e9f250525)
[4b8ee08](https://github.com/nocodb/nocodb/pull/7181/commits/4b8ee083f5689fd543f5a727c98a384e9f250525)

File: packages/nc-gui/assets/style.scss (hunk 6 above)
Comment (musharaf-nocodb) [reply packages/nc-gui/assets/style.scss]: **Fixed in below commit**

[fix : removed unwanted spaces](https://github.com/nocodb/nocodb/pull/7181/commits/4b8ee083f5689fd543f5a727c98a384e9f250525)
[4b8ee08](https://github.com/nocodb/nocodb/pull/7181/commits/4b8ee083f5689fd543f5a727c98a384e9f250525)

File: packages/nc-gui/assets/style.scss (hunk 4 above)
Comment (musharaf-nocodb) [reply packages/nc-gui/assets/style.scss]: **Fixed in below commit**

[fix : removed unwanted spaces](https://github.com/nocodb/nocodb/pull/7181/commits/4b8ee083f5689fd543f5a727c98a384e9f250525)
[4b8ee08](https://github.com/nocodb/nocodb/pull/7181/commits/4b8ee083f5689fd543f5a727c98a384e9f250525)

File: packages/nc-gui/assets/style.scss [hunk 8]
```diff
@@ -591,7 +604,7 @@ a {
 }
 
 .ant-popover-inner {
-  padding: 0 !important;
+  @apply rouded-lg !p-0;
```
Comment (wingkwong) [inline packages/nc-gui/assets/style.scss:607]: typo - `rounded-lg`

Comment (musharaf-nocodb) [reply packages/nc-gui/assets/style.scss]: **FIXED in Below Commit**

[fix : fixed typo](https://github.com/nocodb/nocodb/pull/7181/commits/40025173e0006f2acefa2a9bd0d10647fd554b25)
 
[4002517](https://github.com/nocodb/nocodb/pull/7181/commits/40025173e0006f2acefa2a9bd0d10647fd554b25)

**Review by wingkwong — APPROVED** [review]

**Pull request merged**